### PR TITLE
[NPU] Remove template in ext wrapper and fuse functions

### PR DESCRIPTION
**Pull request by XinWangIntel**
### Details:
 - *Remove template in zero_ext_graph_wrappers*
 - *Remove zero_ext_graph_wrappers_interface.hpp*
 - *Add more low level debug log*
 - *Update level-zero-ext repo commit to use 1.9 version*

### Tickets:
 - *156387*

### CODE REVIEW

**Comment by XinWangIntel** [comment]
build_jenkins

File: src/plugins/intel_npu/src/utils/include/intel_npu/utils/zero/zero_graph_ext_wrappers.hpp [hunk 1]
```diff
@@ -0,0 +1,71 @@
+// Copyright (C) 2018-2024 Intel Corporation
```
Comment (pereanub) [inline src/plugins/intel_npu/src/utils/include/intel_npu/utils/zero/zero_graph_ext_wrappers.hpp:1]: Why move it to utils? It is needed only for the graph that is part of the compiled model, right?

**Review by pereanub — COMMENTED** [review]
Btw, can we also update graph_ext_version to 1.9 in the plugin with this PR?

File: src/plugins/intel_npu/src/utils/src/zero/CMakeLists.txt [hunk 2]
```diff
@@ -24,7 +24,7 @@ target_include_directories(${TARGET_NAME}
 
 target_link_libraries(
   ${TARGET_NAME}
-  PUBLIC openvino::runtime openvino::runtime::dev)
+  PUBLIC openvino::runtime openvino::runtime::dev openvino::npu_al)
```
Comment (pereanub) [inline src/plugins/intel_npu/src/utils/src/zero/CMakeLists.txt:27]: Doesn't look good to ad npu_al to utils.

File: src/plugins/intel_npu/src/utils/include/intel_npu/utils/zero/zero_graph_ext_wrappers.hpp (hunk 1 above)
Comment (XinWangIntel) [reply src/plugins/intel_npu/src/utils/include/intel_npu/utils/zero/zero_graph_ext_wrappers.hpp:1]: the utils will be wrapper for zero relate things, and we can add tests easily for ext with this. But put it inside compiler adapter also work for me

File: src/plugins/intel_npu/src/utils/src/zero/CMakeLists.txt (hunk 2 above)
Comment (XinWangIntel) [reply src/plugins/intel_npu/src/utils/src/zero/CMakeLists.txt:27]: Yes, the deps here is NetworkMeta inside npu_al, If we put this wrapper back to compiler_adapter, do not need to make strange deps here. @pereanub , do you see other regressions? I think we still need other ticket to decouple NetworkMeta with ext wrapper later, can move this back to adapter now to not make blocker to that one

**Review by pereanub — APPROVED** [review]

**Review by PatrikStepan — APPROVED** [review]

**Pull request merged**